src/GoogleFit/FitOnFhir.GoogleFit/Client/Handlers/GoogleFitAuthorizationHandler.cs

```diff
@@ -118,37 +113,23 @@ private async Task<IActionResult> Callback(RoutingRequest request)
             }
         }
 
-        private async Task<IActionResult> Revoke(RoutingRequest request)
+        private async Task<IActionResult> Authorize(RoutingRequest request, AuthState state)
         {
-            AuthState state = null;
-
-            var isValidated = await _tokenValidationService.ValidateToken(request.HttpRequest, request.Token);
+            AuthUriResponse response = await _authService.AuthUriRequest(JsonConvert.SerializeObject(state), request.Token);
```

Comment:
I realize that the original implementation did not include this, but this call probably needs to be wrapped in a try/catch that returns a 500 ObjectResult if an error occurs.